azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java

```diff
@@ -135,12 +135,25 @@ private void uploadLogFile(final DatabaseTransOperator transOperator, final int
     }
   }
 
-  int removeExecutionLogsByTime(final long millis)
+  int removeExecutionLogsByTime(final long millis, final int recordCleanupLimit)
+      throws ExecutorManagerException {
+    int totalRecordsRemoved = 0;
+    int removedRecords;
+    do {
+      removedRecords = removeExecutionLogsBatch(millis, recordCleanupLimit);
+      logger.debug("Removed batch of execution logs. Count of records removed in this batch: "
```

Comment:
logger.info?

Reply:
If there are 2million records, I don't want to fill logs with this line. That is why I kept it as debug.